### PR TITLE
number_of_replicas

### LINKED ISSUE #482: number_of_replicas
Hunting for indices with `number_of_replicas` being set to zero

They might cause data loss in case of node crash, so want to alert on such indices 

From what have I seen it seems that the exporter just ignores `number_of_replicas` even so it is being retrieved because of missing mapping [here](https://github.com/prometheus-community/elasticsearch_exporter/blob/master/collector/indices_settings_response.go#L30)

Wondering if this one can be added

### PR DESCRIPTION
**Pull request by mac2000**
motivation: hunting for indices with `number_of_replicas` being set to zero, which might cause data loss in certain scenarios, need this metric to configure alert

output will be:

```
# HELP elasticsearch_indices_settings_number_of_replicas index setting number_of_replicas
# TYPE elasticsearch_indices_settings_number_of_replicas gauge
elasticsearch_indices_settings_number_of_replicas{index="foo"} 3
elasticsearch_indices_settings_number_of_replicas{index="bar"} 0
```

closes #482

### CODE REVIEW

File: collector/indices_settings.go [hunk 1]
```diff
@@ -90,6 +90,21 @@ func NewIndicesSettings(logger log.Logger, client *http.Client, url *url.URL) *I
 					return val
 				},
 			},
+			{
+				Type: prometheus.GaugeValue,
+				Desc: prometheus.NewDesc(
+					prometheus.BuildFQName(namespace, "indices_settings", "number_of_replicas"),
```
Comment (SuperQ) [inline collector/indices_settings.go:96]: "number_of" is redundant. We can simplify this.

```suggestion
					prometheus.BuildFQName(namespace, "indices_settings", "replicas"),
```

**Review by SuperQ — CHANGES_REQUESTED** [review]
Minor nit, otherwise LGTM.

**Comment by mac2000** [comment]
sorry for a noob question but seems like I have stuck with this DCO whatever, is there a chance you can press the button or should I figure out how to fix this check?

**Comment by rccrdpccl** [comment]
Any chance we can get this merged @SuperQ ? I would love to have this metric exported

**Comment by SuperQ** [comment]
This needs a DCO sign-off. You can use `git commit -s --amend` to add it.

**Comment by mac2000** [comment]
@SuperQ thank you, seems like it did the trick

File: README.md [hunk 2]
```diff
@@ -162,6 +162,7 @@ Further Information
 | elasticsearch_indices_segments_memory_bytes                           | gauge     | 1           | Current memory size of segments in bytes
 | elasticsearch_indices_settings_stats_read_only_indices                | gauge     | 1           | Count of indices that have read_only_allow_delete=true
 | elasticsearch_indices_settings_total_fields                           | gauge     |             | Index setting value for index.mapping.total_fields.limit (total allowable mapped fields in a index)
+| elasticsearch_indices_settings_number_of_replicas                     | gauge     |             | Index setting value for index.number_of_replicas
```
Comment (sysadmind) [inline README.md:165]:
```suggestion
| elasticsearch_indices_settings_replicas                     | gauge     |             | Index setting value for index.replicas
```

**Review by sysadmind — CHANGES_REQUESTED** [review]
The README missed the update of the metric name.

**Comment by mac2000** [comment]
@SuperQ  @sysadmind is there any chance it will be merged somewhere in the future or should something else be done?

**Review by sysadmind — APPROVED** [review]

**Comment by sysadmind** [comment]
LGTM. Thanks!

**Pull request merged**